mcs/class/System/Test/System.Net.Sockets/SocketTest.cs

```diff
@@ -2600,6 +2600,67 @@ public void SendGeneric ()
 		}
 
 		[Test]
+		public void ConcurrentExceedSocketLimit ()
+		{
+			var tasks = new Task[4];
+			for (int i = 0; i < 4; i++) {
+				tasks[i] = Task.Factory.StartNew (() => SendGenericExceedBuffer ());
+			}
+			Task.WaitAll (tasks);
+		}
+
+		[Test]
+		public void SendGenericExceedBuffer ()
+		{
+			// Create a buffer larger than the default max.
```

Comment:
If I am not mistaken, a `send` might block forever if the sender OS buffer is full, and the receiver also has a full buffer and is not reading anything from it. So, shouldn't we have 2 Task, one sending and one receiving, to avoid this kind of deadlock?